fabtests/common/shared.c

```diff
@@ -770,9 +779,49 @@ int ft_accept_next_client() {
 			return ret;
 	}
 
+	/*
+	 * Considering the next client is created in a separate process, the oob
+	 * sock needs to be re-initialized for the new client before the server
+	 * exchanges its address with the new client.
+	 * ft_init_oob closes the listen socket right after the server accepts
+	 * the client's connection once. So when the server wants to accept the
+	 * next client in the oob channel, it needs to re-initialize the listen
+	 * socket and do the accept again. Before doing that, the connection
+	 * socket (oob_sock) needs to be closed and reset to -1 so the ft_init_oob()
+	 * can establish a new connection.
+	 */
+	if (opts.options & FT_OPT_OOB_ADDR_EXCH) {
+		if (oob_sock != -1) {
+			ret = ft_close_oob();
+			if (ret) {
+				FT_PRINTERR("ft_close_oob", ret);
+				return ret;
+			}
+		}
+		ret = ft_init_oob();
+		if (ret) {
+			FT_PRINTERR("ft_init_oob", ret);
+			return ret;
+		}
```

Comment:
Move this code into a new function, ft_reset_oob().  I could see how other tests might adopt a similar flow.